x-pack/legacy/plugins/uptime/public/state/effects/index_status.ts

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { takeLatest } from 'redux-saga/effects';
+import { indexStatusAction } from '../actions';
+import { fetchIndexStatus } from '../api';
+import { fetchEffectFactory } from './fetch_effect';
+
+export function* fetchIndexStatusEffect() {
+  yield takeLatest(
+    indexStatusAction.get,
+    fetchEffectFactory(fetchIndexStatus, indexStatusAction.success, indexStatusAction.fail)
```

Comment:
```ts
fetchEffectFactory(fetchIndexStatus, indexStatusAction.success, indexStatusAction.fail)
```
I like this `get`, `success`, `fail` pattern.

Reply:
yes, it really makes usage simple.